pkg/services/notes.go

```diff
@@ -31,3 +31,26 @@ func NewFetchNotes(
 		return fetchBySystemIntakeID(ctx, id)
 	}
 }
+
+// NewCreateNote is a service to create and return a new note
+// associated with a given SystemIntake
+func NewCreateNote(
+	config Config,
+	create func(context.Context, *models.Note) (*models.Note, error),
+	authorize func(context.Context) (bool, error),
+) func(context.Context, *models.Note) (*models.Note, error) {
+	return func(ctx context.Context, note *models.Note) (*models.Note, error) {
+		ok, err := authorize(ctx)
+		if err != nil {
+			return nil, err
+		}
+		if !ok {
+			return nil, &apperrors.ResourceNotFoundError{
+				Err:      errors.New("failed to authorize create note"),
+				Resource: models.Note{},
+			}
+		}
+
+		return create(ctx, note)
```

Comment:
I think an error returned here should be wrapped in a QueryError (unless it's already wrapped in pkg/storage)

Reply:
I understand some of these words... I'll try and snag some of your time to chat about this one.